arm-mediaservices/2015-10-01/swagger/media.json

```diff
@@ -509,13 +575,13 @@
           "maxLength": 24
         },
         "type": {
-          "description": "Specifies the type of the resource.",
+          "description": "The type of the resource - mediaservices.",
           "$ref": "#/definitions/ResourceType"
         }
       }
     },
     "ResourceType": {
-      "description": "Type of MediaService resource used in CheckNameAvailability.",
+      "description": "Type of MediaService resource.",
       "readOnly": true,

```

Comment:
type is a required property for CheckNameAvailabilityInput. However its definition that is "ResourceType" says that it is readOnly. "readOnly" in swagger means that this property is server populated (in other words it wont be sent on the wire during serialization). Thus having something required and marking it as readOnly for models that will be used as body parameters causes conflict of interest. Please remove readOnly from "ResourceType".